source/features/rgh-improve-new-issue-form.tsx

```diff
@@ -5,6 +5,27 @@ import * as pageDetect from 'github-url-detection';
 import features from '.';
 import clearCacheHandler from '../helpers/clear-cache-handler';
 import {isRefinedGitHubRepo} from '../github-helpers';
+import {expectToken, expectTokenScope} from '../github-helpers/api';
+
+async function validateToken(): Promise<void> {
+	try {
+		await expectToken();
+	} catch {
+		throw new Error('Refined GitHub needs a Personal Access Token to work properly.');
```

Comment:
A link to the options page here would be nice.

Actually, it'd be nice to be able to drop a special link/placeholder to create a link to the options page (and maybe even a "Clear cache" button), in any issue conversation on the RGH repo.

Reply:
> it'd be nice to be able to drop a special link/placeholder

We're still in the extension context, it's easy to create this:


```js
function OptionsButton(): JSX.Element {
	return <button onClick={() => browser.runtime.openOptionsPage()}>Options la la la</a>
}
```

and then just use `select('yo').append(<OptionsButton/>)

Reply:
Actually we already kind of have this, in `rgh-welcome-issue`. It just needs to be generalized to any issue conversation.

Reply:
That's a bit different because the welcome issue exists without the extension so that piece of code is just _extending_ the page. Here instead we're adding a whole block of text.

Technically we can do the same for the "Clear cache" button though:


```js
// rgh-improve-new-issue-form
select('input[id*="extension_cache"]')!.parentElement!.after(
		<ClearCacheButton/>,
	);
```


```
//options.tsx
select('p#clear-cache')!.append(
		<ClearCacheButton/>,
	);
```